aws/resource_aws_lb_listener_rule_test.go

```diff
@@ -829,3 +879,190 @@ resource "aws_security_group" "alb_test" {
   }
 }`, lbName, targetGroupName)
 }
+
+func testAccAWSLBListenerRuleConfig_autoPriorityBase(lbName, targetGroupName string) string {
+	return fmt.Sprintf(`
+resource "aws_lb_listener" "front_end" {
+   load_balancer_arn = "${aws_lb.alb_test.id}"
+   protocol = "HTTP"
+   port = "80"
+
+   default_action {
+     target_group_arn = "${aws_lb_target_group.test.id}"
+     type = "forward"
+   }
+}
+
+resource "aws_lb" "alb_test" {
+  name            = "%s"
+  internal        = true
+  security_groups = ["${aws_security_group.alb_test.id}"]
+  subnets         = ["${aws_subnet.alb_test.*.id}"]
+
+  idle_timeout = 30
+  enable_deletion_protection = false
+
+  tags {
+    Name = "TestAccAWSALB_basic"
+  }
+}
+
+resource "aws_lb_target_group" "test" {
+  name = "%s"
+  port = 8080
+  protocol = "HTTP"
+  vpc_id = "${aws_vpc.alb_test.id}"
+
+  health_check {
+    path = "/health"
+    interval = 60
+    port = 8081
+    protocol = "HTTP"
+    timeout = 3
+    healthy_threshold = 3
+    unhealthy_threshold = 3
+    matcher = "200-299"
+  }
+}
+
+variable "subnets" {
+  default = ["10.0.1.0/24", "10.0.2.0/24"]
+  type    = "list"
+}
+
+data "aws_availability_zones" "available" {}
+
+resource "aws_vpc" "alb_test" {
+  cidr_block = "10.0.0.0/16"
+
+  tags {
+    Name = "TestAccAWSALB_basic"
+  }
+}
+
+resource "aws_subnet" "alb_test" {
+  count                   = 2
+  vpc_id                  = "${aws_vpc.alb_test.id}"
+  cidr_block              = "${element(var.subnets, count.index)}"
+  map_public_ip_on_launch = true
+  availability_zone       = "${element(data.aws_availability_zones.available.names, count.index)}"
+
+  tags {
+    Name = "TestAccAWSALB_basic"
+  }
+}
+
+resource "aws_security_group" "alb_test" {
+  name        = "allow_all_alb_test"
+  description = "Used for ALB Testing"
+  vpc_id      = "${aws_vpc.alb_test.id}"
+
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags {
+    Name = "TestAccAWSALB_basic"
+  }
+}`, lbName, targetGroupName)
+}
+
+func testAccAWSLBListenerRuleConfig_autoPriorityFirst(lbName, targetGroupName string) string {
+	return testAccAWSLBListenerRuleConfig_autoPriorityBase(lbName, targetGroupName) + fmt.Sprintf(`
+resource "aws_lb_listener_rule" "first" {
+  listener_arn = "${aws_lb_listener.front_end.arn}"
+
+  action {
+    type = "forward"
+    target_group_arn = "${aws_lb_target_group.test.arn}"
+  }
+
+  condition {
+    field = "path-pattern"
+    values = ["/first/*"]
+  }
+}
+
+resource "aws_lb_listener_rule" "third" {
```

Comment:
We test a static rule priority with the `_autoPriorityStatic` configuration below. We should either just use this rule here and remove the static configuration below or remove this rule here.

Reply:
It was the leftover when I tried to use the gap priority in the beginning. I think it's still useful here as test case of existing priorities with gap in between. And as explained above, static priority here is not intended to test static priority but to test switching `priority` argument.

Reply:
Valid point 👍